packages/cli-core/API.md

```diff
@@ -64,6 +64,9 @@ export class Format {
 // @public (undocumented)
 export const format: Format;
 
+// @public
+export const getPackageManagerName: () => string;

```

Comment:
Should this be a method on `PackageManagerController` ?

Reply:
It can, but that would require some refactor in the `cli-core` method as it's currently  built for installing _with_ package managers. I also see that these methods would likely end up being static inside the `PackageManagerController`.

Reply:
Maybe it's an abstraction that isn't super important, but we have tried pretty hard to not expose actual package manager names outside of the PackageManagerController. That class already has `runWithPackageManager`, maybe it needs `runCtrlCCallback` which the YarnClassicPackageManagerController can treat as a noop but the other ones can run the callback

Reply:
I see.
I'm a bit concerned that internals of `PackageManagerController` become open and we start to spread `if` conditions with hardcoded package manager names, i.e. escaping the abstraction.

Perhaps instead of `if (getPackageManagerName() === 'yarn-classic') {`
we should have something like `if (!this.packageManagerController.supportPromptingAfterSignal) {`

That way information about `yarn` never leaks into sandbox and we use abstracted flag to drive the behavior.

Reply:
Makes sense, update the PR.